### PR TITLE
[TRY] Add API calls for the new gateway setup API.

### PR DESCRIPTION
**Pull request by samueljseay**
Raising the PR here first for discussion before a PR to the main repo

### CODE REVIEW

File: classes/class-kco-gateway.php [hunk 1]
```diff
@@ -396,6 +396,37 @@ private function check_if_eu( $store_base_location ) {
...
+			return __( 'Use API username and API password you downloaded in the Klarna Merchant Portal. Don’t use your email address.', 'klarna-checkout-for-woocommerce' );
+		}
+
+		/**
+		 * Get the oAuth connection URL.
+		 *
+		 * @param string $return_url The URL to return to after the oAuth connection has been established.
+		 * @return string Connection URL.
+		 */
+		public function get_oauth_connection_url( $return_url = '' ) {
```
Comment (joshuatf) [inline classes/class-kco-gateway.php]: Noted in Slack, but I think we can drop this method.

File: classes/class-kco-gateway.php [hunk 2]
```diff
@@ -396,6 +396,37 @@ private function check_if_eu( $store_base_location ) {
 			}
 		}
 
+		/**
+	 	 * Get required setting keys for setup.
+		 *
+		 * @return array Array of setting keys used for setup.
+		 */
+		public function get_required_settings_keys() {
+			$setting_suffix = $this->check_if_eu( $store_base_location['country'] ) === 'EU' ? 'eu' : 'us';
```
Comment (joshuatf) [inline classes/class-kco-gateway.php:406]: `$store_base_location` is undefined here which I think is the root cause of [this comment](https://github.com/woocommerce/woocommerce-admin/pull/7085#pullrequestreview-674370705).

```suggestion
			$store_base_location = wc_get_base_location();
			$setting_suffix = $this->check_if_eu( $store_base_location['country'] ) === 'EU' ? 'eu' : 'us';
```

**Review by joshuatf — COMMENTED** [review]
Thanks so much for putting this together, @samueljseay!

Left a couple quick fixes.  We can also go ahead and add `needs_setup` to determine if setup has been finished yet:

```php
		public function needs_setup() {
			$store_base_location = wc_get_base_location();
			$setting_suffix      = $this->check_if_eu( $store_base_location['country'] ) === 'EU' ? 'eu' : 'us';
			
			return ! $this->get_option( 'merchant_id_' . $setting_suffix ) || ! $this->get_option( 'shared_secret_' . $setting_suffix );
		}
```

File: classes/class-kco-gateway.php (hunk 2 above)
Comment (samueljseay) [reply classes/class-kco-gateway.php:406]: Ah thanks so much for spotting this @joshuatf !

File: classes/class-kco-gateway.php (hunk 1 above)
Comment (samueljseay) [reply classes/class-kco-gateway.php]: Resolved https://github.com/samueljseay/klarna-checkout-for-woocommerce/pull/1/commits/bc9afe0ce5f014ac43c3ee384c1978dc1835ef21

**Pull request closed without merging**